docs/source/configuration/header-propagation.mdx

```diff
@@ -68,14 +68,58 @@ Enables you to selectively remove headers that were included in the client's req
 
 ### `insert`
 
-Enables you to add custom headers to requests going to a specific subgraph. These headers are always static strings that originate in the router, instead of originating in the client.
+Enables you to add custom headers to requests going to a specific subgraph. These headers are strings (statics, coming from request body or from context) that originate in the router, instead of originating in the client.
+
+- Insert static header
 
 ```yaml
 - insert:
     name: "sent-from-our-apollo-router"
     value: "indeed"
 ```
 
+- Insert header from context
+
+```yaml
+- insert:
+    name: "sent-from-our-apollo-router-context"
+    from_context: "my_key_in_context"
+```
+
+- Insert header from request body
+
+```yaml
+- insert:
+    name: "sent-from-our-apollo-router-request-body"
+    path: ".operationName" # It's a JSON path query to fetch the operation name from request body
+    default: "UNKNOWN" # If no operationName has been specified
+```
+
+> Example JSON path queries
+>
+> Let's say you have a JSON request body with the following structure:
+>
+> ```json
+> {
+>   "items": [
+>     {
+>       "unwanted": 7,
+>       "wanted": { "x": 3, "y": 7 },
+>       "array": [3, 2, 1]
+>     },
+>     {
+>       "isImportant": true
+>     }
+>   ]
+> }
+> ```
+>
+> To fetch the value of the field `isImportant`, the corresponding path is `.items[1].isImportant`.
+>
+> To fetch the value of the field `x`, the corresponding path is `.items[0].wanted.x`.
+>
+> JSON path queries always begin with a period `.`
```

Comment:
Suggest adding an example request and response so that users are able to construct their paths.